不支持Docker直接部署吗

不支持Docker直接部署吗？

@jwcesign 
mac上的docker desktop是不支持docker run --network host 选项的，所以会导致默认的安装脚本里面安装的127.0.0.1在容器中被认为是容器自己的循环地址。所以在mac上安装时需要改一下安装脚本(hack/install_kubespider.sh)：

```bash

#!/bin/sh

set -e

# 1.Load env and echo logo
source hack/env.sh

echo "[INFO] Start to deploy with default configuration..."

# 2.Check env
ret=`docker version`
if [[ $? != 0 ]]; then
    echo "[ERROR] Please install docker"
    exit 1
fi

# 3.Create necessary directory
mkdir -p ${KUBESPIDER_HOME}/kubespider/nas/
mkdir -p ${KUBESPIDER_HOME}/kubespider/aria2/

# 4.Set registry
source hack/util.sh
util::set_registry_for_image

# 5.Deploy aria2
docker run -d \
    --name aria2-pro \
    --restart unless-stopped \
    --log-opt max-size=1m \
    # 去掉这一行，并手动映射端口
    ## --network host \
    -p 6800:6800 \
    -p 6888:6888 \
    -e PUID=$UID \
    -e PGID=$GID \
    -e RPC_SECRET=kubespider \
    -e RPC_PORT=6800 \
    -e LISTEN_PORT=6888 \
    -v ${KUBESPIDER_HOME}/kubespider/aria2/:/config \
    -v ${KUBESPIDER_HOME}/kubespider/nas/:/downloads/ \
    ${image_registry}/aria2-pro:latest

# 6.Deploy kubespider
export KUBESPIDER_DEFAULT_VERSION="latest"
if [[ ${KUBESPIDER_VERSION} == "" ]]; then
    export KUBESPIDER_VERSION=${KUBESPIDER_DEFAULT_VERSION}
fi
docker run -itd --name kubespider \
    -v ${KUBESPIDER_HOME}/kubespider/.config:/app/.config \
    -e PUID=$UID \
    -e PGID=$GID \
    # 去掉这一行，并手动映射端口
    ## --network=host \
    -p 3080:3080 \
    --restart unless-stopped \
    ${image_registry}/kubespider:${KUBESPIDER_VERSION}

# 7.Give necessary info
echo "[INFO] Deploy successful, check the information:"
echo "*******************************************"
echo "Kubespider config path: ${KUBESPIDER_HOME}/kubespider/.config"
echo "Download file path: ${KUBESPIDER_HOME}/kubespider/nas/"
echo "Kubespider webhook address: http://<server_ip>:3080"
echo "Aria2 server address: http://<server_ip>:6800/jsonrpc, you can use any gui or webui to connect it"
echo "Aria2 default secret is:kubespider"
echo "*******************************************"

```

等容器跑起来之后，再进入aria2容器中使用ip addr 命令查看docker分配的ip地址，然后将这个ip地址修改到配置文件中，然后重启容器就好了，chrome插件的server地址也需要用这个ip地址

> @jwcesign 
> mac上的docker desktop是不支持docker run --network host 选项的，所以会导致默认的安装脚本里面安装的127.0.0.1在容器中被认为是容器自己的循环地址。所以在mac上安装时需要改一下安装脚本(hack/install_kubespider.sh)：
> 
> ```bash
> 
> #!/bin/sh
> 
> set -e
> 
> # 1.Load env and echo logo
> source hack/env.sh
> 
> echo "[INFO] Start to deploy with default configuration..."
> 
> # 2.Check env
> ret=`docker version`
> if [[ $? != 0 ]]; then
>     echo "[ERROR] Please install docker"
>     exit 1
> fi
> 
> # 3.Create necessary directory
> mkdir -p ${KUBESPIDER_HOME}/kubespider/nas/
> mkdir -p ${KUBESPIDER_HOME}/kubespider/aria2/
> 
> # 4.Set registry
> source hack/util.sh
> util::set_registry_for_image
> 
> # 5.Deploy aria2
> docker run -d \
>     --name aria2-pro \
>     --restart unless-stopped \
>     --log-opt max-size=1m \
>     # 去掉这一行，并手动映射端口
>     ## --network host \
>     -p 6800:6800 \
>     -p 6888:6888 \
>     -e PUID=$UID \
>     -e PGID=$GID \
>     -e RPC_SECRET=kubespider \
>     -e RPC_PORT=6800 \
>     -e LISTEN_PORT=6888 \
>     -v ${KUBESPIDER_HOME}/kubespider/aria2/:/config \
>     -v ${KUBESPIDER_HOME}/kubespider/nas/:/downloads/ \
>     ${image_registry}/aria2-pro:latest
> 
> # 6.Deploy kubespider
> export KUBESPIDER_DEFAULT_VERSION="latest"
> if [[ ${KUBESPIDER_VERSION} == "" ]]; then
>     export KUBESPIDER_VERSION=${KUBESPIDER_DEFAULT_VERSION}
> fi
> docker run -itd --name kubespider \
>     -v ${KUBESPIDER_HOME}/kubespider/.config:/app/.config \
>     -e PUID=$UID \
>     -e PGID=$GID \
>     # 去掉这一行，并手动映射端口
>     ## --network=host \
>     -p 3080:3080 \
>     --restart unless-stopped \
>     ${image_registry}/kubespider:${KUBESPIDER_VERSION}
> 
> # 7.Give necessary info
> echo "[INFO] Deploy successful, check the information:"
> echo "*******************************************"
> echo "Kubespider config path: ${KUBESPIDER_HOME}/kubespider/.config"
> echo "Download file path: ${KUBESPIDER_HOME}/kubespider/nas/"
> echo "Kubespider webhook address: http://<server_ip>:3080"
> echo "Aria2 server address: http://<server_ip>:6800/jsonrpc, you can use any gui or webui to connect it"
> echo "Aria2 default secret is:kubespider"
> echo "*******************************************"
> 
> ```
> 
> 等容器跑起来之后，再进入aria2容器中使用ip addr 命令查看docker分配的ip地址，然后将这个ip地址修改到配置文件中，然后重启容器就好了，chrome插件的server地址也需要用这个ip地址

大佬可以提个pr来适配一下macos吗？

--- a/hack/install_kubespider.sh
+++ b/hack/install_kubespider.sh
@@ -23,10 +23,8 @@ source hack/util.sh
 util::set_registry_for_image
 
 # 5.Set os related args
-aria2_os_related_arg=""
 kubespider_os_related_arg=""
 if [[ "$(uname)" == "Darwin" ]]; then
-    aria2_os_related_arg="-p 6800:6800 -p 6888:6888"
     kubespider_os_related_arg="-p 3080:3080"
 else
     kubespider_os_related_arg="--network=host"
@@ -37,7 +35,8 @@ docker run -d \
     --name aria2-pro \
     --restart unless-stopped \
     --log-opt max-size=1m \
-    $aria2_os_related_arg \
+    -p 6800:6800 \
+    -p 6888:6888 \
     -e PUID=$UID \
     -e PGID=$GID \
     -e RPC_SECRET=kubespider \
@@ -70,7 +69,8 @@ if [[ "$(uname)" == "Darwin" ]]; then
         echo "[WARN] Can not find the aria2-pro container IPAddress. Please check your aria2 container IPAddress and modify your download_provider.yaml manually"
     else
         cd ${KUBESPIDER_HOME}/kubespider/.config
-        sed -ig "21s#http.*#http://$aria2_container_ip#g" download_provider.yaml
+        line=$(awk '/aria2/{flag=1; next} flag && /rpc_endpoint_host/{print NR; exit}' download_provider.yaml)
+        sed -i "" "${line}s#http.*#http://$aria2_container_ip#g" download_provider.yaml
     fi
 fi
 

--- a/hack/install_kubespider.sh
+++ b/hack/install_kubespider.sh
@@ -29,7 +29,6 @@ if [[ "$(uname)" == "Darwin" ]]; then
     aria2_os_related_arg="-p 6800:6800 -p 6888:6888"
     kubespider_os_related_arg="-p 3080:3080"
 else
-    aria2_os_related_arg="--network=host"
     kubespider_os_related_arg="--network=host"
 fi
 

--- a/hack/install_kubespider.sh
+++ b/hack/install_kubespider.sh
@@ -22,12 +22,23 @@ mkdir -p ${KUBESPIDER_HOME}/kubespider/aria2/
 source hack/util.sh
 util::set_registry_for_image
 
-# 5.Deploy aria2
+# 5.Set os related args
+aria2_os_related_arg=""
+kubespider_os_related_arg=""
+if [[ "$(uname)" == "Darwin" ]]; then
+    aria2_os_related_arg="-p 6800:6800 -p 6888:6888"
+    kubespider_os_related_arg="-p 3080:3080"
+else
+    aria2_os_related_arg="--network=host"
+    kubespider_os_related_arg="--network=host"
+fi
+
+# 6.Deploy aria2
 docker run -d \
     --name aria2-pro \
     --restart unless-stopped \
     --log-opt max-size=1m \
-    --network host \
+    $aria2_os_related_arg \
     -e PUID=$UID \
     -e PGID=$GID \
     -e RPC_SECRET=kubespider \
@@ -37,7 +48,7 @@ docker run -d \
     -v ${KUBESPIDER_HOME}/kubespider/nas/:/downloads/ \
     ${image_registry}/aria2-pro:latest
 
-# 6.Deploy kubespider
+# 7.Deploy kubespider
 export KUBESPIDER_DEFAULT_VERSION="latest"
 if [[ ${KUBESPIDER_VERSION} == "" ]]; then
     export KUBESPIDER_VERSION=${KUBESPIDER_DEFAULT_VERSION}
@@ -46,11 +57,25 @@ docker run -itd --name kubespider \
     -v ${KUBESPIDER_HOME}/kubespider/.config:/app/.config \
     -e PUID=$UID \
     -e PGID=$GID \
-    --network=host \
+    $kubespider_os_related_arg \
     --restart unless-stopped \
     ${image_registry}/kubespider:${KUBESPIDER_VERSION}
 
-# 7.Give necessary info
+# 8.Wait 5s for containers creating
+sleep 5
+
+# 9.Change download provider config IPAddress when OS is MacOS
+if [[ "$(uname)" == "Darwin" ]]; then
+    aria2_container_ip=$(docker inspect aria2-pro | grep IPAddress | sed -n '2p' | awk '{print substr($2, 2, length($2)-3)}')
+    if [[ -z "$aria2_container_ip" ]]; then
+        echo "[WARN] Can not find the aria2-pro container IPAddress. Please check your aria2 container IPAddress and modify your download_provider.yaml manually"
+    else
+        cd ${KUBESPIDER_HOME}/kubespider/.config
+        sed -ig "21s#http.*#http://$aria2_container_ip#g" download_provider.yaml
+    fi
+fi
+
+# 10.Give necessary info
 echo "[INFO] Deploy successful, check the information:"
 echo "*******************************************"
 echo "Kubespider config path: ${KUBESPIDER_HOME}/kubespider/.config"

